extensions/warp-ipfs/src/js_exports.rs

```diff
@@ -0,0 +1,180 @@
+use std::str::FromStr;
+
+use crate::WarpIpfs;
+use js_sys::Uint8Array;
+use warp::crypto::DID;
+use warp::multipass::{
+    identity::{self, Identity, IdentityProfile},
+    MultiPass,
+};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct MultiPassBox {
+    inner: Box<dyn MultiPass>,
+}
+#[wasm_bindgen]
+impl MultiPassBox {
+    pub fn new(instance: WarpIpfs) -> Self {
+        Self {
+            inner: Box::new(instance.clone()) as Box<_>,
+        }
+    }
+    pub async fn create_identity(
+        &mut self,
+        username: Option<String>,
+        passphrase: Option<String>,
+    ) -> Result<IdentityProfile, JsError> {
+        self.inner
+            .create_identity(username.as_deref(), passphrase.as_deref())
+            .await
+            .map_err(|e| e.into())
+    }
+
+    pub async fn get_identity(
+        &self,
+        id_variant: Identifier,
+        id_value: JsValue,
+    ) -> Result<JsVecIdentity, JsError> {
```

Comment:
We could turn the `Vec<Identity>` into `JsValue` with `serde-wasm-bindgen`, could we not?

Reply:
nice. didn't know about `serde-wasm-bindgen`. I made the change now. Also, worth mentioning `wasm-bindgen` has recently merged in support for `Vec<T>` in async functions but have not yet released a new version. When the next version is out we won't need `serde-wasm-bindgen` anymore for this particular case.